apps/src/templates/instructions/TeacherFeedback.jsx

```diff
@@ -107,6 +107,24 @@ class TeacherFeedback extends Component {
     const {user, serverLevelId, teacher} = this.props;
     const {studentId} = this.state;
 
+    window.addEventListener('beforeunload', event => {
+      const latestFeedback =
+        this.state.latestFeedback.length > 0
+          ? this.state.latestFeedback[0]
+          : null;
+      const feedbackUnchanged =
```

Comment:
you're also calculating `feedbackUnchanged` [here](https://github.com/code-dot-org/code-dot-org/blob/98901122be9cc3465461da387eb049ff611ad3c8/apps/src/templates/instructions/TeacherFeedback.jsx#L211-L216), so we should move this logic into a function in the component so that it only lives in one place

Reply:
Good idea! Will do!